File: app/src/main/java/org/example/example/GameEngineClassic.java
package org.example.example;

import edu.austral.dissis.chess.gui.*;
import org.example.example.fabric.GameFabricator;
import org.example.example.game.*;
import org.example.example.movement.InfiniteMove;
import org.example.example.movement.MovementBehaviour;
import org.example.example.piece.Color;
import org.example.example.piece.Piece;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class GameEngineClassic implements GameEngine {
    Game game;
    List<ChessPiece> pieces;
    PlayerColor playerColor;

    public GameEngineClassic(){
       this.game = GameFabricator.createClassicGame();
       pieces = updateChessPieceList();
    }

    public List<ChessPiece> updateChessPieceList(){
        List<ChessPiece> pieceChess = new ArrayList<>();
        List<Cordinate> cordinateWithPieces = this.game.getBoard().getAllCordinatesWithPieces();
        for (int i = 0; i < cordinateWithPieces.size() ; i++) {
            Piece piece = cordinateWithPieces.get(i).getPiece();
            pieceChess.add(new ChessPiece(Integer.toString(i), piece.getColor().equals(Color.WHITE) ? PlayerColor.WHITE : PlayerColor.BLACK, new Position(cordinateWithPieces.get(i).getX(), cordinateWithPieces.get(i).getY()),  piece.getType().toString()));
        }
        return pieceChess;
    }

    @NotNull
    @Override
    public MoveResult applyMove(@NotNull Move move) {
        game.makeAMove(move.component1().component1() , move.component1().component2() , move.component2().component1(), move.component2().component2());
        game.changeTurn();
        return new NewGameState(updateChessPieceList(), game.getTurn().equals(Color.WHITE) ? PlayerColor.WHITE : PlayerColor.BLACK);
    }

    @NotNull
    @Override
    public InitialState init() {
        return new InitialState(new BoardSize(8, 8), pieces, PlayerColor.WHITE );
    }
}

File: app/src/main/java/org/example/GameEngineClassic.kt
package org.example

import edu.austral.dissis.chess.gui.*
import org.example.fabric.GameFabricator
import org.example.game.piece.Color

class GameEngineClassic  : GameEngine {

    var game = GameFabricator().createClassicGame()

    override fun applyMove(move: Move): MoveResult {
        try {
            if (game.isItCheckMate()) {
                if (game.getTurn() == Color.WHITE) {
                    return GameOver(PlayerColor.WHITE)
                }else{
                    return GameOver(PlayerColor.BLACK)
                }
            }
            game = game.movePiece(move.from.column, move.from.row, move.to.column, move.to.row)
            if  (game.getTurn() == Color.WHITE) {
                return NewGameState(chessPieces() , PlayerColor.WHITE)
            }else{
                return NewGameState(chessPieces() , PlayerColor.BLACK)
            }
        }catch (e : Exception){
            return InvalidMove( e.message.toString())
        }
    }

    override fun init(): InitialState {
        return InitialState(BoardSize(game.getBoard().getBoardWidth(), game.getBoard().getBoardHeight()), chessPieces(), PlayerColor.WHITE)
    }

    fun chessPieces(): List<ChessPiece>{
        val listChessPieces = ArrayList<ChessPiece>()
        for (i in game.getBoard().getAllPieces()){
                val cordinate = game.getBoard().getCordinateByPieceId(i.id)
                if (i.color == Color.WHITE){
                    listChessPieces.add(ChessPiece(i.id.toString(), PlayerColor.WHITE , Position(cordinate.y , cordinate.x), i.type.toString()))
                }else{
                    listChessPieces.add(ChessPiece(i.id.toString(), PlayerColor.BLACK , Position(cordinate.y , cordinate.x), i.type.toString()))
                }
        }
        return listChessPieces
    }
}